### PR TITLE
Remove support for old rubies and rails

### PR DESCRIPTION
**Pull request by gssbzn**
All rubies before 2.3 are EOL
All Rails before 5.0 are EOL

Some clean up and improvements

### CODE REVIEW

**Comment by gssbzn** [comment]
I tried to add Rails 5.2 on travis but it's failing for the schema dump, I'm open to suggestions on how to fix those as it's taking me lot of time getting around the code for this

**Comment by gssbzn** [comment]
This also addresses the changes of #71 that looks abandoned

File: lib/hair_trigger.rb [hunk 1]
```diff
@@ -141,29 +137,28 @@ def generate_migration(silent = false)
...
-      prefix = ActiveRecord::VERSION::STRING < "3.1." ? "self." : ""
-      File.open(file_name, "w"){ |f| f.write <<-MIGRATION }
+      File.open(file_name, "w") { |f| f.write <<-RUBY }
 # This migration was auto-generated via `rake db:generate_trigger_migration'.
 # While you can edit this file, any changes you make to the definitions here
 # will be undone by the next auto-generated trigger migration.
 
-class #{migration_name} < ActiveRecord::Migration
-  def #{prefix}up
+class #{migration_name} < ActiveRecord::Migration[#{ActiveRecord::VERSION::MAJOR}.#{ActiveRecord::VERSION::MINOR}]
```
Comment (barthez) [inline lib/hair_trigger.rb]: @gssbzn You could use `ActiveRecord::Migration.current_version` here.

**Comment by barthez** [comment]
Hey @jenseng,

Could you take a look at it? It would be nice to have properly generated migration files finally.

Thanks!

**Comment by barthez** [comment]
@gssbzn  I found a reason why Rails 5.2 fails on travis. You need to update [these lines](https://github.com/jenseng/hair_trigger/blob/master/lib/hair_trigger/schema_dumper.rb#L102-L104) to `class_attribute :previous_schema`. 

The reason behind it is simple, in Rails 5.2 schema dumper started using Adapter pattern. Each adapter inherits from `ActiveRecord::SchemaDumper`. In specs we assign `ActiveRecord::SchemaDumper.previous_schema`.  The `attr_accessor` macro creates non-inheritable accessor, in opposite to `class_attribute` where accessors inherits values from parent classes.

**Comment by gssbzn** [comment]
@barthez thanks for the help there, once I get a free moment later today I'll try to add those changes 👍

**Comment by barthez** [comment]
You should probably update `README.md` file as it mentions supported Ruby and Rails versions.

**Comment by gssbzn** [comment]
@barthez done, let's hope @jenseng can review and merge soon

**Comment by barthez** [comment]
@gssbzn You missed compability section at the bottom of readme file. Otherwise everything looks perfect!

**Comment by gssbzn** [comment]
@jenseng is there anything preventing this getting merge?

**Comment by tyandell** [comment]
Can this be merged in and released? It would be great to have this for compatibility with Rails 5.2.

**Comment by jenseng** [comment]
Thanks @gssbzn and sorry for the delay!

**Comment by jenseng** [comment]
Released in [0.2.22](https://rubygems.org/gems/hairtrigger/versions/0.2.22)

**Pull request merged**